tools/wptrunner/wptrunner/executors/executormarionette.py

```diff
@@ -673,29 +679,18 @@ def do_test(self, test):
     def do_testharness(self, protocol, url, timeout):
         parent_window = protocol.testharness.close_old_windows(protocol)
 
-        if timeout is not None:
-            timeout_ms = str(timeout * 1000)
-        else:
-            timeout_ms = "null"
-
         if self.protocol.coverage.is_enabled:
             self.protocol.coverage.reset()
 
-        format_map = {"abs_url": url,
-                      "url": strip_server(url),
-                      "window_id": self.window_id,
-                      "timeout_multiplier": self.timeout_multiplier,
-                      "timeout": timeout_ms,
-                      "explicit_timeout": timeout is None}
-
-        script = self.script % format_map
-
-        protocol.base.execute_script(script, async=True)
-        test_window = protocol.testharness.get_test_window(self.window_id, parent_window)
+        format_map = {"url": strip_server(url)}
 
+        protocol.base.execute_script("window.open('about:blank', '%s', 'noopener')" % self.window_id)
+        test_window = protocol.testharness.get_test_window(self.window_id, parent_window,
+                                                           timeout=10*self.timeout_multiplier)
+        self.protocol.base.set_window(test_window)
```

Comment:
This seems a bit needlessly complex now we're opening a new window. Can we not: navigate to about:blank (to avoid the window potentially opening anything else), close everything open, then navigate that (with `<a rel=noopener>` probably, rather than opening another new window?) to the test?

Reply:
That seems far more complex? What's the benefit?

Reply:
It avoids guessing what window we need to keep open. Which is mostly a hypothetical concern it turns out, admittedly.